### PR TITLE
fix(PropertyBindingParser): properly parse event bindings with assigements

### LINKED ISSUE #981: Event bindings should allow assignments
This code

```
<input type="number" [value]="number" (keyup)="number = $event.target.value">
```

Throws: 
```
Uncaught (in promise) Error: Parser Error: Binding expression cannot contain assignments at column 8 in [number = $event.target.value] in null
```

But it should work since assignments should be supported in event binding (not in data binding)

### PR DESCRIPTION
**Pull request by pkozlowski-opensource**
Fixes #981

### CODE REVIEW

File: modules/angular2/test/core/compiler/pipeline/property_binding_parser_spec.js [hunk 1]
```diff
@@ -68,13 +68,22 @@ export function main() {
       // "(click[])" is not an expected syntax and is only used to validate the regexp
       results = createPipeline().process(el('<div (click[])="b()"></div>'));
       expect(MapWrapper.get(results[0].eventBindings, 'click[]').source).toEqual('b()');
+    });
 
+    it('should detect () syntax with assigements in expressions', () => {
```
Comment (vsavkin) [inline modules/angular2/test/core/compiler/pipeline/property_binding_parser_spec.js]: I am not sure if this test name actually reflects what you are trying to test. Assignments do not affect the detection of the () or on- syntax.

Comment (pkozlowski-opensource) [inline modules/angular2/test/core/compiler/pipeline/property_binding_parser_spec.js]: True. It should be more `"should detect () syntax and properly parse expressions with assignments"` or something. Any other suggestions? (sigh, naming things _is_ hard :-))

Comment (vsavkin) [inline modules/angular2/test/core/compiler/pipeline/property_binding_parser_spec.js]: "should detect () syntax and properly parse expressions with assignments" is good. Maybe "should parse event handlers as actions" or something? Cause it is not just assignments. Pipes, for instance, do not work in event handlers.

**Comment by pkozlowski-opensource** [comment]
@vsavkin I've renamed tests and rebased the PR. Do you have any additional comments or is it good to go?

**Pull request closed without merging**